File: Data Structures & Algorithms/balanced-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func isBalanced(root *TreeNode) bool {
    res := true
    var dfs func(*TreeNode) int
    dfs = func(curr *TreeNode) int {
        if (curr == nil) {
            return 0
        }

        left := dfs(curr.Left)
        right := dfs(curr.Right)
        diff := right - left
        if (math.Abs(float64(diff)) > 1) {
            res = false
        }

        return 1 + max(left, right)
    }

    dfs(root)
    return res
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    res: int = 0
    def goodNodes(self, root: TreeNode) -> int:
        self.dfs(root, root.val)
        return self.res
                    
    def dfs(self, curr: TreeNode, biggest_seen: int):
        if not curr:
            return
                
        if curr.val >= biggest_seen:
            self.res += 1
            self.dfs(curr.right, curr.val)
            self.dfs(curr.left, curr.val)
        else:
            self.dfs(curr.right, biggest_seen)
            self.dfs(curr.left, biggest_seen)